### PR TITLE
Added support for multi-controlled gates on qiskit

### PR DESCRIPTION
**Pull request by golanor**
Added support for multi-controlled gates on qiskit, see #387

### CODE REVIEW

File: tangelo/linq/tests/test_translator_circuit.py [hunk 1]
```diff
@@ -180,6 +180,26 @@ def test_qiskit(self):
         sim_results = qiskit_simulator.run(translated_circuit).result()
         np.testing.assert_array_almost_equal(sim_results.get_statevector(translated_circuit), reference_big_msq, decimal=6)
 
+    @unittest.skipIf("qiskit" not in installed_backends, "Test Skipped: Backend not available \n")
+    def test_qiskit_multi_control(self):
+        from tangelo.linq import Gate, Circuit, translate_circuit
```
Comment (ValentinS4t1qbit) [inline tangelo/linq/tests/test_translator_circuit.py:185]: All of these have been imported already at the beginning of the module, I think they can be removed (L25)

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
Thank you, this looks good ! You even supported an extra gate ! The test is convincing. 

- I think some imports can be cleaned up.
- One of the checks currently fails (`pycodestyle` says theres a trailing white space somewhere). Some other checks fail but I don't think it is related to your contribution and we know why.

Out of curiosity: would considering the CRX, CRY, etc as MCRX, MCRY, ... with a single control qubit be possible ? This would simplify the code, and then we may not even need the `crx`, `cry`, `crz`, gate from Qiskit in the dictionary.

It may be that if a user passes a single integer in the Tangelo gate, the software automatically converts it into a list of 1 element. In that case, the the full list of control qubits can always be passed in the translation.

If I am wrong and this becomes tricky, we can forget about this improvement.

**Comment by golanor** [comment]
> Out of curiosity: would considering the CRX, CRY, etc as MCRX, MCRY, ... with a single control qubit be possible ? This would simplify the code, and then we may not even need the `crx`, `cry`, `crz`, gate from Qiskit in the dictionary.
> 
> It may be that if a user passes a single integer in the Tangelo gate, the software automatically converts it into a list of 1 element. In that case, the the full list of control qubits can always be passed in the translation.
> 
> If I am wrong and this becomes tricky, we can forget about this improvement.

It depends on the end realization - the compilation of the MCR() gates is wasteful compared to the CR() gates. If that doesn't matter then we can probably change it.

Regarding the trailing whitespace - I haven't found where it is happening, or which action fails. Can you please point me to where to find it?

**Comment by ValentinS4t1qbit** [comment]
> It depends on the end realization - the compilation of the MCR() gates is wasteful compared to the CR() gates. If that doesn't matter then we can probably change it.

Then I do not mind leaving them alone, the code is still pretty simple. Thanks for the insight.

> Regarding the trailing whitespace - I haven't found where it is happening, or which action fails. Can you please point me to where to find it?

Unexpected: in an unrelated file outside of your PR, I see a whitespace in the printing format. Our linter complains about it.
Would you please remove that whitespace ? This way all tests should pass and we're calling this PR done !

../tangelo/algorithms/variational/adapt_vqe_solver.py:322:61: E203 whitespace before ':'

I saw that while checking the list of checks at the bottom of this PR page, and clicking "details".
Out of curiosity, are you able to click and access the details of the checks ? It would be troublesome if our users could not see them.

Screenshot 2024-05-29 at 6 26 56 PM

Thank you so much, appreciate your patience on this. Unexpected things happen :)

**Comment by golanor** [comment]
> Unexpected: in an unrelated file outside of your PR, I see a whitespace in the printing format. Our linter complains about it. Would you please remove that whitespace ? This way all tests should pass and we're calling this PR done !
> 
> ../tangelo/algorithms/variational/adapt_vqe_solver.py:322:61: E203 whitespace before ':'
> 
> I saw that while checking the list of checks at the bottom of this PR page, and clicking "details". Out of curiosity, are you able to click and access the details of the checks ? It would be troublesome if our users could not see them.

> Thank you so much, appreciate your patience on this. Unexpected things happen :)

I only now realized I can scroll down the checks and see the ones that failed... 

Thank you for being so available! I fixed the whitespace issue.

File: tangelo/linq/tests/test_translator_circuit.py [hunk 2]
```diff
@@ -180,6 +180,26 @@ def test_qiskit(self):
         sim_results = qiskit_simulator.run(translated_circuit).result()
         np.testing.assert_array_almost_equal(sim_results.get_statevector(translated_circuit), reference_big_msq, decimal=6)
 
+    @unittest.skipIf("qiskit" not in installed_backends, "Test Skipped: Backend not available \n")
+    def test_qiskit_multi_control(self):
+        from tangelo.linq import translate_circuit
```
Comment (ValentinS4t1qbit) [inline tangelo/linq/tests/test_translator_circuit.py:185]: This function was already imported as `translate_c` earlier in the file, we can just reuse that and remove this line :)

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
A small final change and the bounty is yours ! 

Your import was already available as a slightly different name, we do not need imports inside that function (in general we prefer to import at the beginning of the module, rather than re-importing inside each function that needs it).

The checks have all passed, so I'm sure it will all work out after your next change.

**Review by ValentinS4t1qbit — APPROVED** [review]
Looks good to me !

Thank you for your contribution, I hope this was an interesting experience.
The devil's in the details, and details accumulate over time and make open-source codebase challenging... thank you for your patience :)

If you think Tangelo is an interesting project, please don't hesitate to give us a little ⭐ a consider mentioning it or using it :)

**Pull request merged**